### PR TITLE
Verify V2 email

### PR DESCRIPTION
**Pull request by SecondeJK**
This PR adds the ability to register an email for use with Verify v2

# New 

* Added a new endpoint to the Account API, in order to register an email address
* Added a new tab to Verify V2 workflows for sending a request using email.

# Checklist

- [x] version number incremented (in the `info` section of the spec)

### CODE REVIEW

File: definitions/verify.v2.yml [hunk 1]
```diff
@@ -449,6 +482,25 @@ components:
...
+      type: string
+      minLength: 1
+      maxLength: 50
+      example: '447700400080'
+      description: An optional sender number, in the [E.164](https://en.wikipedia.org/wiki/E.164) format. Don't use a leading + or 00 when entering a phone number, start with the country code, for example, 447700900000.
+
+    ToEmail:
+      type: string
+      format: email
+      example: 'john.bull@company.com'
```
Comment (pardel) [inline definitions/verify.v2.yml:495]: can we use Alice and Bob as names? https://en.wikipedia.org/wiki/Alice_and_Bob

File: definitions/verify.v2.yml [hunk 2]
```diff
@@ -393,6 +390,42 @@ components:
...
+                    $ref: '#/components/schemas/ToEmail'
+                  from:
+                    $ref: '#/components/schemas/FromEmail'
+                required:
+                  - channel
+                  - to
+              example: [
+                  {
+                    "channel": "email",
+                    "to": "lemony.snicket@acme.com",
```
Comment (pardel) [inline definitions/verify.v2.yml:422]: can we use Alice, Bob & co for names? https://en.wikipedia.org/wiki/Alice_and_Bob

File: definitions/account.yml [hunk 3]
```diff
@@ -590,6 +640,37 @@ components:
...
+            The delivery method by which the value would be assigned. Always `email`
+          type: string
+          format: enum
+          example: email
+        value:
+          description: >-
+            The email adress to attach to the application(s)
+          type: string
+          format: email
+          example: lemony.snicket@acme.com
```
Comment (pardel) [inline definitions/account.yml:660]: Let's keep it consistent and use @company.com for email addresses. Also, let's use standard names: https://en.wikipedia.org/wiki/Alice_and_Bob

File: definitions/account.yml [hunk 4]
```diff
@@ -430,6 +458,28 @@ components:
       schema:
         type: string
   schemas:
+    registerEmailResponse:
+      type: object
+      properties:
+        name:
+          type: string
+          description: Secret ID
```
Comment (pardel) [inline definitions/account.yml:466]: Is this a secret?

File: definitions/account.yml [hunk 5]
```diff
@@ -430,6 +458,28 @@ components:
...
+          example: ['6ec9eb7de1d34e77ac2c0ad597c9554a', '5dba5838689f402dba0c77957e2181f3']
+        value:
+          type: string
+          description: Value given to the provider to attach to the Application IDs.
+          example: alice@company.com
+        provider:
+          type: string
+          format: enum
+          description: Enum identifer of provider type.
+          example: "email"
```
Comment (pardel) [inline definitions/account.yml:482]: Do we need to specify what values the enum can be?

File: definitions/account.yml [hunk 6]
```diff
@@ -430,6 +458,28 @@ components:
       schema:
         type: string
   schemas:
+    registerEmailResponse:
+      type: object
+      properties:
+        name:
+          type: string
+          description: Secret ID
+          example: my-email-link
```
[the remark this answers is not in the document]
Comment (SecondeJK) [reply definitions/account.yml:467]: Damn, good spot. This shouldn't be here

File: definitions/account.yml (hunk 4 above)
Comment (SecondeJK) [reply definitions/account.yml:466]: No, an optional name parameter, I expect this is a copy paste error

File: definitions/account.yml [hunk 7]
```diff
@@ -590,6 +640,37 @@ components:
...
+      required:
+        - provider
+        - value
+      properties:
+        provider:
+          description: >-
+            The delivery method by which the value would be assigned. Always `email`
+          type: string
+          format: enum
+          example: email
```
[the remark this answers is not in the document]
Comment (SecondeJK) [reply definitions/account.yml:654]: Yes this is missing enum definition, will change it

**Review by pardel — APPROVED** [review]

**Review by pardel — APPROVED** [review]

**Pull request merged**